server/src/main/java/org/opensearch/identity/ServiceAccount.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+import java.util.List;
+
+/**
+ * This interface defines a ServiceAccount.
+ *
+ * A ServiceAccount is a principal which can retrieve its associated permissions.
+ */
+public interface ServiceAccount extends Principal {
+
+    public List<String> getPermissions();
```

Comment:
Working backwards, where would these permissions be defined? One of the primary reasons plugins stash the threadcontext (i.e. switching context to be able to do anything) is to write to a system index. What does the plugin/extension's relationship to the system index look like and how is it configured? Is it through `SystemIndex.getSystemIndexDescriptors` or is it configured elsewhere in a yml file?

Reply:
Hi @cwperks, great question! 

My initial thought for permission definition would be in the yml file for any plugins and defined in the extension installation request for extensions. The goal here is to eventually be able to offer some form of in-core FGAC as well as a clear boundary between what does and does not require impersonation. We will also be able to use the context switching to manage when we request tokens for authc/z. 

That being said, I am still working on the way I think permissions may look so this is not set in stone.